nossomercadolivre-api/src/main/java/com/github/nossomercadolivre/Category.java

```diff
@@ -0,0 +1,77 @@
+package com.github.nossomercadolivre;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.util.Objects;
+
+import static javax.persistence.GenerationType.IDENTITY;
+
+@Entity
+@Table
+public class Category {
+
+    @Id
+    @GeneratedValue(strategy = IDENTITY)
+    @Column(name = "ID_CATEGORY")
+    private Long id;
+
+    @NotBlank
+    @Size(max = 100)
+    @Column(nullable = false, unique = true, length = 100)
+    private String name;
+
+    @JoinColumn(name = "ID_CATEGORY_MOTHER", referencedColumnName = "ID_CATEGORY")
+    @ManyToOne(optional = true, fetch = FetchType.LAZY)
+    private Category categoryMother;
+
+    /**
+     * Just to framework
+     */
+    protected Category() {
+    }
+
+    public Category(@NotNull @NotEmpty @Size(max = 100) final String name) {
+        this.name = name;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Category getCategoryMother() {
```

Comment:
*Não retornamos nulo dentro das regras da aplicação. Pense que seu computador vai explodir.*

E se a categoria mãe for nula?

Reply:
Tem alguma sugestão? Costumo encontrar resistência a escrever código diferente disto.

Em SmallTalk, fazia um NullObject, mas ai o objeto precisaria ter o método isNull, isso signfica que preciso ter um if em algum lugar pra ver se é null mesmo usando o NullObject

Reply:
Aqui eu sugiro retornar Optional<Category>

Reply:
Uhn...... nunca pensei em usar assim!! Me parece uma boa ideia, vou experimentar!